src/rgw/rgw_sal_motr.cc

```diff
@@ -1294,51 +1294,64 @@ MotrObject::~MotrObject() {
 //    return read_op.prepare(dpp);
 //  }
 
-int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx, Attrs* setattrs, Attrs* delattrs, optional_yield y, rgw_obj* target_obj)
+int MotrObject::fetch_obj_entry_and_key(const DoutPrefixProvider* dpp, rgw_bucket_dir_entry& ent, std::string& bname, std::string& key, rgw_obj* target_obj)
```

Comment:
Some code optimization done in https://github.com/Seagate/cortx-rgw/pull/221/files (in main branch)
Better do on similar lines, to avoid major conflicts later, when we merge feature to main branch

Reply:
code optimization for set_obj_attrs() and get_obj_attrs() is done in fetch_obj_entry_and_key()
+ some additional changes are also done like you asked in previous comment.
@selvakumaar5496 please confirm the optimization changes you did in https://github.com/Seagate/cortx-rgw/pull/221/files are covered in this PR. If yes we can discard the changes in https://github.com/Seagate/cortx-rgw/pull/221/files and keep these changes.

Reply:
yeah i can see changes in this feature branch are much more equipped for versioned objects handling, like get_bucket_dir_ent() here. however i feel we will have merge changes in these PRs, like 1305 to 1311 in here collected in read_bucket_info() PR #221

Reply:
Agree with you @cdeshmukh